### PR TITLE
Player may disable receiving /msg

### PR DESCRIPTION
**Pull request by TheMolkaPL**

### CODE REVIEW

File: src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java [hunk 1]
```diff
@@ -102,6 +103,22 @@ public void sendAdmin(Match match, MatchPlayer sender, @Nullable @Text String me
...
+    MatchPlayer matchReceiver = manager.getPlayer(receiver);
+    if (matchReceiver != null) {
+      SettingValue option = matchReceiver.getSettings().getValue(SettingKey.PM);
+
+      switch (option) {
+        case PM_OFF:
+          if (!sender.getBukkit().hasPermission(Permissions.STAFF)) {
+            String name = receiver.getDisplayName(sender.getBukkit()) + ChatColor.RED;
+            Component component =
+                new PersonalizedTranslatable("command.message.blockedNoPermissions", name);
```
Comment (Pablete1234) [inline src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java]: I'm unsure what blockedNoPermissions message exactly is, but it shouldn't be a no perms, but rather a "player doesn't want DMS"

File: src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java [hunk 2]
```diff
@@ -102,6 +103,22 @@ public void sendAdmin(Match match, MatchPlayer sender, @Nullable @Text String me
       desc = "Send a direct message to a player",
       usage = "[player] [message]")
   public void sendDirect(Match match, MatchPlayer sender, Player receiver, @Text String message) {
+    MatchPlayer matchReceiver = manager.getPlayer(receiver);
+    if (matchReceiver != null) {
+      SettingValue option = matchReceiver.getSettings().getValue(SettingKey.PM);
+
+      switch (option) {
+        case PM_OFF:
+          if (!sender.getBukkit().hasPermission(Permissions.STAFF)) {
```
Comment (Pablete1234) [inline src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java]: A switch-case for 1 option, feels pretty redundant
```java
if (option == PM_OFF && !sender.getBukkit().hasPermission(Permissions.STAFF)) {
  // send deny
  return;
}
```

File: src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java (hunk 1 above)
Comment (TheMolkaPL) [reply src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java]: It was already in the strings.

https://github.com/Electroid/PGM/blob/1a604f129d510489aa9ec0a398a31b3c787c139d/src/main/i18n/templates/strings.properties#L818

File: src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java (hunk 2 above)
Comment (TheMolkaPL) [reply src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java]: Yeah, I'll change that.

File: src/main/java/tc/oc/pgm/api/setting/SettingKey.java [hunk 3]
```diff
@@ -17,7 +17,8 @@
   CHAT("chat", CHAT_TEAM, CHAT_GLOBAL, CHAT_ADMIN), // Changes the default chat channel
   DEATH(
       Arrays.asList("death", "dms"), DEATH_ALL, DEATH_OWN), // Changes which death messages are seen
-  PICKER("picker", PICKER_AUTO, PICKER_ON, PICKER_OFF); // Changes when the picker is displayed
+  PICKER("picker", PICKER_AUTO, PICKER_ON, PICKER_OFF), // Changes when the picker is displayed
+  PM(Arrays.asList("pm", "dm", "msg"), PM_ON, PM_OFF); // Changes if private messages are accepted
```
Comment (Electroid) [inline src/main/java/tc/oc/pgm/api/setting/SettingKey.java]: `PM` -> `MESSAGE`. `message` and `dm`

File: src/main/java/tc/oc/pgm/api/setting/SettingValue.java [hunk 4]
```diff
@@ -21,7 +21,9 @@
   DEATH_ALL("death", "all"), // Send all death messages, highlight your own
   PICKER_AUTO("picker", "auto"), // Display after cycle, or with permissions.
   PICKER_ON("picker", "on"), // Display the picker GUI always
-  PICKER_OFF("picker", "off"); // Never display the picker GUI
+  PICKER_OFF("picker", "off"), // Never display the picker GUI
+  PM_ON("pm", "on"), // Always accept private messages
```
Comment (Electroid) [inline src/main/java/tc/oc/pgm/api/setting/SettingValue.java]: private -> direct

**Comment by TheMolkaPL** [comment]
I've resolved conflicts 👌

**Pull request merged**